app/services/media_storage/stored_path.rb

```diff
@@ -1,28 +1,51 @@
 # frozen_string_literal: true
 
 module MediaStorage
+  # StoredPath: module used by AzureAdapter for constructing the
+  # stored path/url of the media resource in azure
+  #
+  # Old S3 paths will include the panoptes-uploads domain prefix
+  # and the environment in their path, for example:
+  # panoptes-uploads.zooniverse.org/staging/subject_location/49f02f969a5.jpeg
+  # These URLs need to be rewritten such that domain prefix and env get removed
+  # Azure native paths do not need to be rewritten
   module StoredPath
-    # Construct the media access URL
-    # and ensure we remove the old aws s3 domain prefix in the stored paths
-    # but leave intact any non old s3 URL, i.e. new azure ones
-    def self.media_url(url, stored_path)
-      uri = URI("https://#{stored_path}")
-      # if the parse succeeds, we know we have a valid domain prefix,
-      # i.e. it is an old s3 stored path - hence, we need to rewrite the url
-      PublicSuffix.parse(uri.host)
-      uri_path = uri.path
+    class << self
+      ENV_PATH_PREFIX = '/' + Rails.env
 
-      env_prefix = '/' + Rails.env
-      # remove env prefix if present
-      uri_path = uri_path.sub(env_prefix, '') if uri_path.start_with? env_prefix
+      def media_path(stored_path)
+        rewrite_stored_path(stored_path)
+      rescue PublicSuffix::DomainNotAllowed
+        # a valid stored path without a TLD prefix
+        stored_path
+      end
 
-      File.join(url, uri_path)
-    rescue PublicSuffix::DomainNotAllowed
-      # failure here indicates we do not have
-      # a valid domain prefix in the stored path
-      # e.g. /user_avatar/1e5fc9b5-86f1-4df3-986f-549f02f969a5.jpeg
-      # so we do not need to rewrite the URL
-      File.join(url, stored_path)
+      def media_url(domain_prefix, stored_path)
+        azure_path = rewrite_stored_path(stored_path)
+        File.join(domain_prefix, azure_path)
+      rescue PublicSuffix::DomainNotAllowed
+        azure_path = stored_path
+        File.join(domain_prefix, azure_path)
```

Comment:
Not blocking at all, can this be written like this?
```suggestion
        begin
          azure_path = rewrite_stored_path(stored_path)
        rescue PublicSuffix::DomainNotAllowed
          azure_path = stored_path
        end
        File.join(domain_prefix, azure_path)
```